test/e2e/autotls/domain_mapping_test.go

```diff
@@ -0,0 +1,125 @@
+// +build e2e
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autotls
+
+import (
+	"context"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/networking/test/conformance/ingress"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/reconciler"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+	v1test "knative.dev/serving/test/v1"
+)
+
+func TestDomainMappingAutoTLS(t *testing.T) {
+	var env config
+	if err := envconfig.Process("", &env); err != nil {
+		t.Fatalf("Failed to process environment variable: %v.", err)
+	}
+
+	t.Parallel()
+	ctx := context.Background()
+
+	clients := e2e.SetupWithNamespace(t, test.TLSNamespace)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "runtime",
+	}
+
+	if len(env.TLSServiceName) != 0 {
+		names.Service = env.TLSServiceName + "dm-tls"
+	}
+
+	// Clean up on test failure or interrupt.
+	test.EnsureTearDown(t, clients, &names)
+
+	// Set up initial Service.
+	svc, err := v1test.CreateServiceReady(t, clients, &names)
+	if err != nil {
+		t.Fatalf("Failed to create initial Service %v: %v", names.Service, err)
+	}
+
+	// Using fixed hostnames can lead to conflicts when multiple tests run at
+	// once, so include the svc name to avoid collisions.
+	host := svc.Service.Name + ".example.com"
```

Comment:
I would suggest using the environment variable [CUSTOM_DOMAIN_SUFFIX](https://github.com/knative/serving/blob/master/test/e2e-auto-tls-tests.sh#L33) as the suffix of the DomainMapping host. (sorry that this is not straightforward).

This is needed for testing HTTP01 challenge as we add the DNS record for the custom domain suffix to point to the ingress IP.